srtcore/srt.h

```diff
@@ -573,9 +573,17 @@ enum SRT_REJECT_REASON
     SRT_REJ_FILTER,      // incompatible packet filter
     SRT_REJ_GROUP,       // incompatible group
 
-    SRT_REJ__SIZE,
+    SRT_REJ_E_SIZE,
 };
 
+// Reject category codes:
+
+#define SRT_REJC_VALUE(code) (1000 * (code/1000))
+#define SRT_REJC_SYSTEM 0     // Codes from above SRT_REJECT_REASON enum
+#define SRT_REJC_SERVER 1000  // Standard server error codes
+#define SRT_REJC_USER 2000    // User defined error codes
```

Comment:
I don't think we need to distinguish SERVER and USER. All these rejection reason codes are user-defined.

Reply:
That's a design-level discussion and we're past that point already.

Reply:
We are exactly at that point now, we are trying to provide a useful API, not just merge something and put a checkbox.

Reply:
So, the merit was the following: the rejection code is something that one application sends and the other - usually completely unrelated application - interprets. Therefore there are reserved two spaces, one for HTTP codes, one for user codes. When an application uses HTTP codes, it should align to the requirements of the HTTP codes so that the other - unrelated - application understands it.

If then an application is using the error code from the "server" range, it must use HTTP codes, or risk being falsely understood as not conforming to the standard. If the application uses the "user" range however, it must make sure - by interchanging some other data using possibly another connection or another medium - that the application "speaks the same language" before specifying any code like this.

In short:

"server" codes: all applications _should_ speak the same language (we can't enforce that of course, but we can at least define a standard).

"user" codes: applications on both sides must be somehow certain, or make sure somehow, that they both speak the same language, and avoid using these codes otherwise (or reject a connection in advance using a HTTP code).